File: encapsulation/src/main/java/blog/yun/encapsulation/product/Order.java
package blog.yun.encapsulation.product;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "product")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Embedded
    private Message message;

    @Column(name = "anti_message_types")
    private String messageTypes;

    @Builder
    public Order(Message message, String messageTypes) {
        this.message = message;
        this.messageTypes = messageTypes;
    }
}

File: spring-kotlin/src/main/kotlin/com/example/springkotlin/domain/order/Order.kt
package com.example.springkotlin.domain.order

import com.example.springkotlin.domain.member.domain.Member
import com.example.springkotlin.domain.transaction.Transaction

import javax.persistence.*

@Entity
@Table(name = "orders")
class Order protected constructor() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0


    @Column(name = "order_number", nullable = false, updatable = false, unique = true)
    lateinit var number: String
        protected set

    @ManyToOne
    @JoinColumn(name = "member_id", nullable = false, updatable = false)
    lateinit var member: Member
        protected set


    @OneToOne
    @JoinColumn(name = "transaction_id", nullable = false, updatable = false)
    lateinit var transaction: Transaction
        protected set

}